docs/edsl.html

```diff
@@ -0,0 +1,661 @@
+<head>
+<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
+<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
+</head>
+
+<body>
+<h1 id="c-tile-edsl">C++ Tile EDSL</h1>
+<p>The C++ Tile EDSL (Embedded Domain Specific Language) provides developers with a
+  way of describing a neural network so that the Stripe-based PlaidML compiler can
+  construct an efficient implementation.</p>
+<p>This tutorial is intended to help machine learning practitioners (or anyone with
+  a background in software engineering and mathematics) get started using the C++
+  Tile EDSL.</p>
+<h2 id="scope-and-warning">Scope and Warning</h2>
+<p>This tutorial provides an introduction to the C++ Tile EDSL. It is intended to
+  help machine learning practitioners get started writing Tile code as quickly as
+  possible, and as such covers core features, not every language detail. This is a
+  tutorial, not a spec, and as such will consist of a series of examples, with a
+  summary reference section at the end.</p>
+<p>This tutorial covers how to use the C++ Tile EDSL, not how Tile code is
+  constructed and manipulated by PlaidML. It does not cover the workings of
+  PlaidML utilities such as the pmlc compiler.</p>
+<p>Tile and PlaidML are still being developed and the APIs discussed here are subject
+  to change.</p>
+<h2 id="how-to-write-tile-code">How to Write Tile Code</h2>
+<h3 id="sum-over-axis">Sum Over Axis</h3>
+<p>We&#39;re ready to look at some C++ Tile code! Here&#39;s an operation that takes the
+  sum over axis <code>0</code> of a 2D tensor (in Keras this would be <code>K.sum(I, axis=0)</code>):</p>
+<pre><code class="lang-c++">Tensor sum_over_axis(const Tensor&amp; I) {
+  TensorDim M, N<span class="hljs-comment">;</span>
+  TensorIndex m, n<span class="hljs-comment">;</span>
+  I.<span class="hljs-keyword">bind_dims(M, </span>N)<span class="hljs-comment">;</span>
+  auto O = TensorOutput(N)<span class="hljs-comment">;</span>
+  O(n) += I(m, n)<span class="hljs-comment">; // contraction</span>
+  return O<span class="hljs-comment">;</span>
+}
+</code></pre>
+<p>An operation such as this which merges together values across one or more
+  indices is called a <em>contraction</em>. The syntax may look a bit odd at first, but
+  it&#39;s related to summation notation. Below we show how this C++ Tile code is
+  related to the mathematical formula for the operation by using colors to
+  highlight corresponding pieces:</p>
+
+<p>
+\[
+\Large
+\textcolor{red}{O[n]}
+\textcolor{yellow}{=}
+\textcolor{green}{\sum_{m}}
+\textcolor{cyan}{I[m, n]}
+\]</p>
+
+<p>
+\[
+\Large
+  \textcolor{red}{O(n)}
+  \textcolor{green}{+=}
```

Comment:
`+=` in code is not rendering well in LaTeX. Try `\,{+}{=}`, I _think_ that should help